File: javascript/28.Find-the-Index-of-the-First-Occurrence-in-a-String.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function (haystack, needle) {
  let index = 0;

  while (index < haystack.length) {
    if (haystack[index] === needle[0]) {
      let p1 = index;
      let p2 = 0;

      while (p2 < needle.length && haystack[p1] === needle[p2]) {
        p1++;
        p2++;
      }

      if (p2 === needle.length) {
        return index;
      }
    }

    index++;
  }

  return -1;
};

File: python/28.Find-the-Index-of-the-First-Occurrence-in-a-String.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        index = 0

        while index < len(haystack):
            if haystack[index] == needle[0]:
                p1 = index
                p2 = 0

                while p2 < len(needle) and p1 < len(haystack) and haystack[p1] == needle[p2]:
                    p1 += 1
                    p2 += 1

                if p2 == len(needle):
                    return index

            index += 1

        return -1